src/components/ChatEntry.js

```diff
@@ -1,22 +1,35 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import {DateTime} from 'luxon';
+import TimeStamp from './TimeStamp';
 
 const ChatEntry = (props) => {
+
+const toggleHeart = () => {
+    props.updateHeart(props);
+  }
+
+  const heart = props.liked ? '❤️' : '🤍';
+
   return (
     <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+      <h2 className="entry-name">{props.sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p>{props.body}</p>
+        <p className="entry-time"><TimeStamp time={props.timeStamp}/></p>
+        <button onClick={toggleHeart} className="like">{heart}</button>
       </section>
     </div>
   );
 };
 
 ChatEntry.propTypes = {
-  //Fill with correct proptypes
+  id: PropTypes.number.isRequired,
+  sender: PropTypes.string.isRequired,
+  body: PropTypes.string.isRequired, 
+  // TimeStamp: PropTypes.DateTime.isRequired,
```

Comment:
It looks like you may have forgotten to bring this back and rename it to `timeStamp`. If this were production code, it would actually be quite important to make sure the shape of `props` is exactly what you expect, because you're passing `props` to `updateHeart` to handle the like-clicking.